sphinx/builders/linkcheck.py

```diff
@@ -66,6 +68,15 @@ def init(self) -> None:
         # set a timeout for non-responding servers
         socket.setdefaulttimeout(5.0)
 
+        if not self.config.linkcheck_allow_unauthorized:
+            deprecation_msg = (
+                "The default value for 'linkcheck_allow_unauthorized' will change "
+                "from `True` in Sphinx 7.3+ to `False`, meaning that HTTP 401 "
+                "unauthorized responses will be reported as broken by default. "
+                "See sphinx-doc/sphinx#11433 for details."
```

Comment:
I know I probably sound like a broken record with this, but I'm still puzzling over exactly the audience we want to warn with this message.

At the moment it's warning people who've opted-in, who presumably have already read the changelog/description.

I'd like to be 99.9% sure about this so that the relevant audiences are informed.

Reply:
Yes, you want to warn people that a feature will be soon deprecated. In other words, you only want to warn people using it (others don't care since they may not be aware of). If you want to warn people reading the changelog but not using the feature, it may be good to add a small comment in the changelog telling them "hey, it's only for 7.3 - 7.x that this feature is supported, so it's really only for a transition".

Now, what you could do for people to make them the transition is actually warn when you encounter a 401 *even if* `linkcheck_allow_unauthorized` is true but using `warnings` and not the logger! that way, people who don't know about the feature will know that in 8.x they need to be careful about this (this is one thing I forgot to mention actually, so my bad).

Reply:
That makes sense, yep.  So for people who already have HTTP 401s occurring during linkchecking, we can inform them that a change-in-behaviour is on the horizon.

Please let me know if that interpretation is correct; I'll prepare that additional warning code locally.  And thanks for being patient, for some reason I've found this warning logic tricky to reason about, and want to get it correct.